Makefile

```diff
@@ -1,26 +1,66 @@
+.PHONY: help venv conda docker docstyle format style black test lint check
+.DEFAULT_GOAL = help
+
+PYTHON = python
+PIP = pip
+CONDA = conda
+SHELL = bash
+
+help:
+	@printf "Usage:\n"
+	@grep -E '^[a-zA-Z_-]+:.*?# .*$$' $(MAKEFILE_LIST) | awk 'BEGIN {FS = ":.*?# "}; {printf "\033[1;34mmake %-10s\033[0m%s\n", $$1, $$2}'
+
+conda:  # Set up a conda environment for development.
+	@printf "Creating conda environment...\n"
+	${CONDA} create --yes --name env python=3.6
+	( \
+	source activate env; \
+	${PIP} install -U pip; \
+	${PIP} install -r requirements.txt; \
+	${PIP} install -r requirements-dev.txt; \
+	conda deactivate; \
+	)
+	@printf "\n\nConda environment created! \033[1;34mRun \`source activate env\` to activate it.\033[0m\n\n\n"
+
+venv:  # Set up a Python virtual environment for development.
+	@printf "Creating Python virtual environment...\n"
+	rm -rf venv
+	${PYTHON} -m venv venv
+	( \
+	source venv/bin/activate; \
+	${PIP} install -U pip; \
+	${PIP} install -r requirements.txt; \
+	${PIP} install -r requirements-dev.txt; \
+	deactivate; \
+	)
+	@printf "\n\nVirtual environment created! \033[1;34mRun \`source venv/bin/activate\` to activate it.\033[0m\n\n\n"
+
+docker:  # Set up a Docker image for development.
+	@printf "Creating Docker image...\n"
+	${SHELL} ./scripts/container.sh --build
+	@printf "\n\nDocker image created! \033[1;34mRefer to pymc4/scripts/README.md for instructions on how to use it.\033[0m\n\n\n"
+
 docstyle:
-	@echo "Checking documentation with pydocstyle..."
+	@printf "Checking documentation with pydocstyle...\n"
 	pydocstyle pymc4/
-	@echo "Pydocstyle passes!\n"
-
+	@printf "\033[1;34mPydocstyle passes!\033[0m\n\n"
 
 format:
-	@echo "Checking code style with black..."
+	@printf "Checking code style with black...\n"
 	black --check pymc4/
-	@echo "Black passes!\n"
+	@printf "\033[1;34mBlack passes!\033[0m\n\n"
 
 style:
-	@echo "Checking code style with pylint..."
+	@printf "Checking code style with pylint...\n"
 	pylint pymc4/
-	@echo "Pylint passes!\n"
+	@printf "\033[1;34mPylint passes!\033[0m\n\n"
 
-black:
+black:  # Format code in-place using black.
```

Comment:
I'm being so super nitpicky here but if we're commenting some of the make recipes we should comment them all. Not a big deal if we don't though since these recipes are really obvious

Reply:
That's what I thought as well! I decided not to document all of them though: if you document it, it will show up when you run `make help`, or just `make`. I only documented the rules that I expected end-users to actually use (in other words, I think developers should focus on "linting", instead of "linting with black" or "linting with pylint"). What do you think?

Reply:
Your explanation is sufficiently convincing for how minor my concern is

I think whatever you do here will be great :)